source/shared/cpp/ObjectModel/MarkDownHtmlGenerator.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+#include <string>
+#include <iomanip>
+#include <vector>
+#include <list>
+#include "BaseCardElement.h"
+
+namespace AdaptiveCards
+{
+
+enum DelimiterType
+{
+    Init,
+    Alphanumeric,
+    Puntuation,
+    Escape,
+    WhiteSpace,
+    Underscore,
+    Asterisk,
+};
+
+// this class knows how to generate html string of their types 
+// - MarkDownStringHtmlGenerator 
+//   it is the most basic form,
+//   it simply retains and return text as string
+// - MarkDownEmphasisHtmlGenerator 
+//   it knows how to handle bold and italic html
+//   tags and apply those to its text when asked to generate html string
+// - MarkDownListHtmlGenerator
+//   it functions simmilary as MarkDownStringHtmlGenerator, but its IsList() returns
+//   true, this is used in generating html block tags
+//   list uses block tag of <ul> all others use <p>
+class MarkDownHtmlGenerator
+{
+    public:
+        MarkDownHtmlGenerator() : m_token(""){};
+        MarkDownHtmlGenerator(std::string &token) : m_token(token){};
+        void MakeItHead() { m_isHead = true; }
+        void MakeItTail() { m_isTail = true; }
+        void MakeItHeadOfNewBlock() { m_isTail = true; }
```

Comment:
>m_isTail [](start = 38, length = 8)

Should be m_isHeadOfNewBlock

Reply:
thanks!

---
In reply to: [157636157](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157636157) [](ancestors = 157636157)